src/System.Text.Encodings.Web/src/System/Text/Encodings/Web/EncoderExtensions.cs

```diff
@@ -0,0 +1,70 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.IO;
+
+namespace System.Text.Encodings.Web
+{
+    /// <summary>
+    /// Helpful extension methods for the encoder classes.
+    /// </summary>
+    public static class EncoderExtensions
+    {
+        /// <summary>
+        /// HTML-encodes a string and writes the result to the supplied output.
+        /// </summary>
+        /// <remarks>
+        /// The encoded value is also safe for inclusion inside an HTML attribute
+        /// as long as the attribute value is surrounded by single or double quotes.
+        /// </remarks>
+        public static void HtmlEncode(this IHtmlEncoder htmlEncoder, string value, TextWriter output)
+        {
+            if (htmlEncoder == null)
+            {
+                throw new ArgumentNullException("htmlEncoder");
```

Comment:
BTW super late to the game here, but are you using C# 6 in these projects? If so, `nameof(htmlEncoder)`. This has saved us so many times on ASP.NET / EF so far!